subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java

```diff
@@ -221,11 +221,11 @@ public void connect(PipedInputStream snk) throws IOException {
     /**
      * Downloads and sets up the JVM arguments for running the Gradle daemon with the file leak detector: https://file-leak-detector.kohsuke.org/
      *
-     * NOTE: This requires running the test with JDK8 and the forking executer.
+     * NOTE: This requires running the test with JDK11 and the forking executer.
```

Comment:
```suggestion
     * NOTE: This requires running the test with at least JDK11 and the forking executer.
```

Are you sure this requires JDK 11 now?

Reply:
Yes, running with -DtestJavaVersion=8 throws:
```
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/kohsuke/file_leak_detector/AgentMain has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Reply:
The problem is that there's no reliable way to know which version of the JDK the build-under-test is using. So our options are:
1. Use a version that supports Java 8, which is 1.14. Assuming it supports everything newer too.
2. Use a version that supports Java 11+ and document it. This is what this PR is doing.
3. Try to guess which version to use and be wrong sometimes

If (1) works for Java 8 and Java 21, I would just do that. If not, I wouldn't bother with the complexity of (3) and just accept this PR as-is. I'd rather the test fixture fail in a known way (does not support Java 8) than for it to sometimes fail in a surprising way.

Reply:
> Use a version that supports Java 8, which is 1.14. Assuming it supports everything newer too.

I think it should work. We have some glitches with Java 17+ support (e.g. https://github.com/jenkinsci/lib-file-leak-detector/issues/86 for sockets), but general functionality should run well

Reply:
I don't think we need to find a perfect solution right now, since this is just very internal integration tests tool.

So I added a change where we use 1.17 for JDK >= 11 else 1.14. Additionally I added also a method:
```
withFileLeakDetection(JavaVersion javaVersion, String... args)
```
if one wants a specific Java version file-leak-detector.

Btw, I tested 1.14 with JDK 17 and JDK crashes and daemon doesn't start.

Reply:
Also all versions on JDK >= 11 report some error in stacktrace like:
```
Caused by: java.lang.UnsupportedOperationException
	at jdk.zipfs/jdk.nio.zipfs.ZipPath.toFile(ZipPath.java:674)
	at org.kohsuke.file_leak_detector.Listener.openDirectoryStream(Listener.java:334)
	... 163 more
```
But the leak detection works it seems.